tests/dialects/test_duckdb.py

```diff
@@ -432,6 +432,12 @@ def test_cast(self):
                 "snowflake": "CAST(COL AS ARRAY)",
             },
         )
+        self.validate_all(
+            "CAST([STRUCT_PACK(a := 1)] AS STRUCT(a BIGINT)[])",
+            write={
+                "SELECT CAST(LIST_VALUE({'a': 1}) AS STRUCT(a BIGINT)[])",
```

Comment:
Might be useful to test array of array here too.

Reply:
Good point, agreed. Feel free to suggest more forms to test, I didn't dig very deep for this one, so maybe I missed something.